### PR TITLE
Created @expo/html-elements

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- [**Docs preview**](https://github.com/expo/expo/blob/@evanbacon/html-elements/init/packages/html-elements/README.md)

# Test Plan

- Added a demo page in NCL `yarn <ios|android|web>` in `apps/bare-expo`.
- Components are tested with `jest-expo-enzyme`.
- Everything is E2E tested with Detox and puppeteer.

### CODE REVIEW

File: packages/html-elements/README.md [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+Simple, light-weight, and well tested, universal semantic HTML elements as React components for iOS, Android, web, and desktop!
+
+We at Expo recommended using platform agnostic primitives like `View`, `Image`, and `Text` whenever possible but sometimes that's not easy! This package aims to solve that while still being an optimal UI package for iOS, and Android.
+
+**What you get**
+
+- Using these components will optimize for accessibility and SEO.
+  - This package takes full advantage of [`react-native-web` a11y rules](https://github.com/necolas/react-native-web/blob/master/packages/docs/src/guides/accessibility.stories.mdx) whenever possible.
+  - The `H1` component will render an `<h1 />` on web, a `UILabel` on iOS, and a `TextView` on Android.
+- Every component excepts styles from the `StyleSheet` API.
```
Comment (brentvatne) [inline packages/html-elements/README.md]:
```suggestion
- Every component accepts styles from the `StyleSheet` API.
```

File: packages/html-elements/README.md [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+- [x] `<tr />` => `<Tr />`
+- [x] `<td />` => `<Td />`
+- [x] `<caption />` => `<Caption />`
+- [x] `<ul />` => `<Ul />`
+- [x] `<ol />` => `<Ol />`
+- [x] `<li />` => `<Li />` or `<FlatList />` from `react-native-web`
+- [ ] `<div />` => `<View />` from `react-native-web`
+- [ ] `<video />` => Use `<Video />` from [`expo-av`](https://docs.expo.io/versions/latest/sdk/video/).
+- [ ] `<img />` => Use `<Image />` from `react-native-web`
+- [ ] `<canvas />` => Use `<GLView />` from [`expo-gl`](https://docs.expo.io/versions/latest/sdk/gl-view/)
```
Comment (brentvatne) [inline packages/html-elements/README.md]: maybe also link to https://github.com/expo/expo-2d-context

File: packages/html-elements/README.md [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+- [ ] `<input type="text" />` => Use `<TextView />`
+- [ ] `<input type="file" />` => Use [`expo-image-picker`](https://docs.expo.io/versions/latest/sdk/imagepicker/) and [`expo-document-picker`](https://docs.expo.io/versions/latest/sdk/document-picker/)
+- [ ] `<iframe />` => On native use `<WebView />` from `@react-native-community/web-view`, this package is not maintained by Expo and doesn't have web support.
```
Comment (brentvatne) [inline packages/html-elements/README.md]: maybe link to this page https://docs.expo.io/versions/latest/sdk/webview/

File: packages/html-elements/README.md [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+
+## Headings
+
+Header elements will use the expected font size and margins from web universally.
```
Comment (brentvatne) [inline packages/html-elements/README.md]: might be good to refer to exactly where this information came from, eg: http://trac.webkit.org/browser/trunk/Source/WebCore/css/html.css if that's where you found it. also would be good to explain how the default styles using ems/rems is translated to native by referring people to the css/units file

**Review by brentvatne — COMMENTED** [review]
nice work @EvanBacon

File: packages/html-elements/src/elements/Lists.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  const elements = React.Children.toArray(children).map((element, index) =>
+    React.cloneElement(element, { bullet: `${index + 1}.` })
+  );
+
+  return <View {...props} style={[styles.ol, props.style]} children={elements} ref={ref} />;
+}) as ComponentType<ViewProps>;
+
+function isTextProps(props: any): props is TextProps {
+  if (props.children !== undefined) {
+    if (typeof props.children === 'string') {
```
Comment (ide) [inline packages/html-elements/src/elements/Lists.tsx]:
```suggestion
    return typeof props.children === 'string';
```

File: packages/html-elements/src/elements/Text.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  return <Text {...props} style={[styles.mark, style]} ref={ref} />;
+}) as ComponentType<TextProps>;
+
+// TODO: Lazy load mono font on native
+export const Code = forwardRef((props: TextProps, ref) => {
+  return <Text {...props} ref={ref} />;
+}) as ComponentType<TextProps>;
+
+export const Strong = B;
+export const Strike = S;
```
Comment (ide) [inline packages/html-elements/src/elements/Text.tsx]: Strike is deprecated, let's just not expose it (del and s instead).

**Review by ide — COMMENTED** [review]
For naming, I'm leaning a bit towards THead, TBody, HR, OL, UL, LI, etc...

File: packages/html-elements/src/css/units.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { PixelRatio, Platform } from 'react-native';
+
+export function rem(pixels: number): number | string {
```
Comment (FiberJW) [inline packages/html-elements/src/css/units.ts]: `pixels` isn't the right variable name. If someone wanted entered 16 "pixels" (which is 1rem, usually), they would instead get 16rem, which is a huge difference. I think renaming `pixels` to `rem/em`, `units` or something similar would clear up any confusion about how to use this.

File: packages/html-elements/src/elements/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1 @@
+export { default as Footer } from '../primitives/View';
```
Comment (FiberJW) [inline packages/html-elements/src/elements/Footer.tsx]: It looks like `contentinfo` would be the appropriate `accessibilityRole` to apply to this element for web. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Contentinfo_role

Comment (FiberJW) [reply packages/html-elements/src/elements/Footer.tsx]: Ah nvm, it looks like you're creating an actual footer element in the file below this. If you were to take the same approach that you did with Article, then the comment above would be relevant.

**Review by brentvatne — APPROVED** [review]

**Pull request merged**